docs/orchestrating-elastic-stack-applications/elasticsearch/autoscaling.asciidoc

```diff
@@ -0,0 +1,197 @@
+
+- Scaling the storage size automatically requires the `ExpandInUsePersistentVolumes` feature to be enabled. It also requires a storage class that supports link:https://kubernetes.io/blog/2018/07/12/resizing-persistent-volumes-using-kubernetes/[volume expansion].
+- Only one persistent volume claim per Elasticsearch node is supported when autoscaling is enabled.
+- Volume size cannot be scaled down.
+
+[float]
+[id="{p}-{page_id}-resources"]
+=== Setting limits
+
+The value set for memory and cpu limits are computed by applying a ratio to the calculated resource request. For memory the default ratio between the request and the limit is 1, meaning that request and limit have the same value. For cpu the default ratio is 0, which means that no limit is set. You can change the default ratio between the request and the limit for both the cpu and memory range using the `requestsToLimitsRatio` field.
+
+For example, you can remove the memory limit and set a cpu limit to twice the value of the request as follows:
+
+[source,json]
+----
+{
+    "policies": [{
+        "name": "di-hot",
+        "roles": ["data_hot", "ingest", "transform"],
+        "resources": {
+            "nodeCount": {
+                "min": 2,
+                "max": 5
+            },
+            "cpu": {
+                "min": 1,
+                "max": 2,
+                "requestsToLimitsRatio": 2
+            },
+            "memory": {
+                "min": "2Gi",
+                "max": "6Gi",
+                "requestsToLimitsRatio": 0
+            }
+        }
+    }]
+}
+----
+
+You can find link:{eck_github}/blob/{eck_release_branch}/config/recipes/autoscaling/elasticsearch.yaml[a complete example in the ECK GitHub repository] which will also show you how to finely tune the autoscaling deciders.
+
+[float]
+[id="{p}-{page_id}-polling-interval"]
+=== Changing the polling interval
+
+The Elasticsearch autoscaling capacity endpoint is polled every minute. This interval duration can be controlled using the `pollingPeriod` field in the autoscaling specification:
+
+[source,json]
+----
+{
+    "pollingPeriod": "42s",
+    "policies": [{
+        "name": "di-hot",
+        "roles": ["data_hot", "ingest", "transform"],
+        "resources": {
+            "nodeCount": {
+                "min": 2,
+                "max": 5
+            },
+            "cpu": {
+                "min": 1,
+                "max": 2
+            },
+            "memory": {
+                "min": "2Gi",
+                "max": "6Gi"
+            }
+        }
+    }]
+}
+----
+
+[float]
+[id="{p}-monitoring"]
+== Monitoring
+
+In addition to the logs generated by the operator an autoscaling status is stored in the `elasticsearch.alpha.elastic.co/autoscaling-status` annotation. The autoscaling status is a JSON document which describes the expected resources for each `NodeSet` managed by an autoscaling policy. It may also contain important messages about the state of the tier.
```

Comment:
```suggestion
In addition to the logs generated by the operator, an autoscaling status is stored in the `elasticsearch.alpha.elastic.co/autoscaling-status` annotation. The autoscaling status is a JSON document which describes the expected resources for each `NodeSet` managed by an autoscaling policy. It may also contain important messages about the state of the tier.
```